packages/pug-lexer/index.js

```diff
@@ -990,172 +1046,244 @@ Lexer.prototype = {
   blockCode: function() {
     var tok
     if (tok = this.scanEndOfLine(/^-/, 'blockcode')) {
-      this.tokens.push(tok);
+      this.tokens.push(this.tokEnd(tok));
       this.interpolationAllowed = false;
       this.callLexerFunction('pipelessText');
       return true;
     }
   },
-
+  
   /**
-   * Attributes.
+   * Attribute Name.
    */
-
-  attrs: function() {
-    if ('(' == this.input.charAt(0)) {
-      var startingLine = this.lineno;
-      this.tokens.push(this.tok('start-attributes'));
-      var index = this.bracketExpression().end
-        , str = this.input.substr(1, index-1);
-
+  
+  attributeName: function(str){
+    var quote = '';
+    var quoteRe = /['"]/;
+    var key = '';
+    var i;
+    
+    characterParser.defaultState();
```

Comment:
This line is a no-op. You don't need to call `defaultState` unless you are going to use the state that gets returned.